slapp/explorer/views.py

```diff
@@ -633,3 +634,18 @@ def basic_charts(request: HttpRequest) -> JsonResponse:
     basic_charts_data = get_basic_charts_data(region)
 
     return JsonResponse(basic_charts_data)
+
+
+def ranges(request: HttpRequest) -> JsonResponse:
+    """Return requested data for ranges on results page."""
+    region = request.GET.get("region", "")
+
+    alternatives = get_alternative_result_for_region(region)
```

Comment:
As mentioned above, I would split up chart data into two chunks and call function with related divergences.